pkg/controller/controlplane/controlplane_controller_test.go

```diff
@@ -110,3 +129,460 @@ func TestReconcile(t *testing.T) {
+						Name:      "test-name",
+						Namespace: "test-namespace",
+					},
+				}),
+				scheme: scheme.Scheme,
+				log:    logging.New(),
+			},
+			args: args{
+				request: reconcile.Request{
+					NamespacedName: types.NamespacedName{Name: "test-name", Namespace: "test-namespace"},
+				},
+			},
+			want:    reconcile.Result{},
+			wantErr: true,
+		},
+		{
+			name: "updates instance status to error if missing EKS cluster",
+			fields: fields{
+				Client: fakeclient.NewFakeClient(
+					&clusterv1alpha1.ControlPlane{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "test-controlplane-name",
+							Namespace: "test-controlplane-namespace",
+							Labels: map[string]string{
+								"eks.owner.name":      "test-eks-name",
+								"eks.owner.namespace": "test-eks-namespace",
+							},
+						},
+					},
+				),
+				scheme: scheme.Scheme,
+				log:    logging.New(),
+			},
+			args: args{
+				request: reconcile.Request{
+					NamespacedName: types.NamespacedName{Name: "test-controlplane-name", Namespace: "test-controlplane-namespace"},
+				},
+			},
+			want:    reconcile.Result{},
+			wantErr: true,
+		},
+		{
+			name: "can describe a controlplane stack",
+			fields: fields{
+				Client: fakeclient.NewFakeClient(
+					&clusterv1alpha1.ControlPlane{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "test-controlplane-name",
+							Namespace: "test-controlplane-namespace",
+							Labels: map[string]string{
+								"eks.owner.name":      "test-eks-name",
+								"eks.owner.namespace": "test-eks-namespace",
+							},
+						},
+					},
+					&clusterv1alpha1.EKS{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "test-eks-name",
+							Namespace: "test-eks-namespace",
+						},
+						Spec: clusterv1alpha1.EKSSpec{
+							ControlPlane: clusterv1alpha1.ControlPlaneSpec{
+								ClusterName: "test-clustername",
+							},
+						},
+					},
+				),
+				scheme: scheme.Scheme,
+				log:    logging.New(),
+				cfnSvc: &fakeaws.MockCloudformationAPI{},
+			},
+			args: args{
+				request: reconcile.Request{
+					NamespacedName: types.NamespacedName{Name: "test-controlplane-name", Namespace: "test-controlplane-namespace"},
+				},
+			},
+			want:    reconcile.Result{RequeueAfter: 5 * time.Second},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &ReconcileControlPlane{
+				Client: tt.fields.Client,
+				scheme: tt.fields.scheme,
+				log:    tt.fields.log,
+				sess:   tt.fields.sess,
+				cfnSvc: tt.fields.cfnSvc,
+			}
+			got, err := r.Reconcile(tt.args.request)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ReconcileControlPlane.Reconcile() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("ReconcileControlPlane.Reconcile() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestReconcileControlPlane_Reconcile_CloudformationStatusPending(t *testing.T) {
```

Comment:
Is it possible to wrap the different Reconcile_CloudformationStatus* into one test with multiple entries in the table?